src/lpython/parser/semantics.h

```diff
@@ -287,4 +288,40 @@ Vec<ast_t*> MERGE_EXPR(Allocator &al, ast_t *x, ast_t *y) {
 #define ATTRIBUTE_REF(val, attr, l) make_Attribute_t(p.m_a, l, \
         EXPR(val), name2char(attr), expr_contextType::Load)
 
+expr_t* CHECK_TUPLE(expr_t *x) {
+    if(is_a<Tuple_t>(*x) && down_cast<Tuple_t>(x)->n_elts == 1) {
+        return down_cast<Tuple_t>(x)->m_elts[0];
+    } else {
+        return x;
+    }
+}
```

Comment:
This function makes sure that the `Tuple` node is not created for a single element.
For ex:
```python
    z = x[1]
```